Week3/homework/app.js

```diff
@@ -0,0 +1,150 @@
+    }
+    res.json(results);
+  });
+};
+
+// Update User
+const updateUser = (req, res) => {
+  const user_id = req.params.id;
+  const user_name = req.body.name;
+  const updateUsersQuery = `UPDATE users SET username = ?  WHERE user_id = ${user_id}`;
+  connection.query(updateUsersQuery, [user_name], err => {
+    if (err) {
+      return res.status(500).send('Something went wrong');
+    }
+    res.status(200).send('user updated');
+  });
+};
+
+// Create todolist
+const createTodoList = (req, res) => {
+  const user_id = req.body.user;
+  const todolist_name = req.body.name;
+  const createTodolist = `INSERT INTO todolist (user_id, todolist_Name) VALUES (?, ?)`;
+  connection.query(createTodolist, [user_id, todolist_name], err => {
+    if (err) {
+      throw err;
+    }
+    res.status(200).send('Todolist created!');
+  });
+};
+
+// Delete todo list
+const deleteTodoList = (req, res) => {
+  const todolist_id = req.params.id;
+  const deleteTodoListQuery = `DELETE FROM todolist WHERE todolist_id = ?`;
+  connection.query(deleteTodoListQuery, [todolist_id], err => {
+    if (err) {
+      return res.status(500).send('Something went wrong');
+    }
+    res.status(200).send('Todo list deleted');
+  });
+};
+
+//Create todos list
+const createTodo = (req, res) => {
+  const todolist_id = req.body.todolist_id;
+  const todo_item = req.body.name;
+  const tag = req.body.tag;
+  const done = req.body.done;
+  const createTodo = `INSERT INTO todos (todolist_id, todo_item, tag, done ) VALUES (?, ?, ?, ?)`;
+  connection.query(createTodo, [todolist_id, todo_item, tag, done], err => {
+    if (err) {
+      return res.status(500).send('Something went wrong');
+    }
+    return res.status(201).send('Todo item created!');
+  });
+};
+
+// Detete todos item
+const deleteTodoItem = (req, res) => {
+  const todo_id = req.params.id;
+  if (!todo_id) {
+    return res.status(400).send('Todo not found!');
+  }
+  const deleteTodoQuery = `DELETE FROM todos WHERE todo_id = ${todo_id}`;
+  connection.query(deleteTodoQuery, [todo_id], err => {
+    if (err) {
+      return res.status(500).send('Something went wrong');
+    }
+    res.status(200).send('Todo item deleted');
+  });
+};
+
+//
+
+//Mark todo item as done
+const markAsDone = (req, res) => {
+  const todo_id = req.params.id;
+  const updateTodoQuery = `UPDATE todos SET done = true  WHERE todo_id = ${todo_id}`;
+  connection.query(updateTodoQuery, [todo_id], err => {
+    if (err) {
+      return res.status(500).send('Something went wrong');
+    }
+    res.status(200).send('Todo item marked as done');
+  });
+};
+
+app.post('/create/user', createUser);
+app.get('/users', getUsers);
+app.get('/users/:id', getUser);
+app.put('/users/:id', updateUser);
+app.post('/create/todolist', createTodoList);
+app.delete('/todolist/:id', deleteTodoList);
+app.post('/create/todo', createTodo);
+app.delete('/todo/:id', deleteTodoItem);
+app.put('/todo/:id', markAsDone);
```

Comment:
For future projects, try to split the code into different modules instead of a single file :)